packages/core/scripts/local/ClaimAllRewards.js

```diff
@@ -1,72 +1,190 @@
+        priceRequests.forEach(({ identifier, time, ancillaryData }, i) => {
+          console.group(`Price Request ${i}`);
+          console.log(`Identifier: ${identifier} (${web3.utils.hexToUtf8(identifier)})`);
+          console.log(`Time: ${time}`);
+          if (ancillaryData) console.log(`Ancillary data: ${ancillaryData}`);
+          console.groupEnd();
+        });
+        console.log(`Total Rewards: ${web3.utils.fromWei(rewards.toString())}`);
+        console.groupEnd();
+      });
+      console.groupEnd();
+    });
+    const rewardSummary = Object.fromEntries(
+      Object.entries(groupedByVoter).map(([voter, rewardArray]) => {
+        const totalRewards = rewardArray.reduce((sum, { rewards }) => sum.add(rewards), toBN("0"));
+        return [voter, web3.utils.fromWei(totalRewards.toString())];
+      })
+    );
+    console.group("Per-voter reward summary:");
+    console.log(rewardSummary);
+    console.groupEnd();
+    return;
+  }
+
+  const dataArray = retrievableRewards.map(({ voter, priceRequests, round, voting }) => {
+    return voting.methods.retrieveRewards(voter, round, priceRequests).encodeABI();
   });
 
-  const valuesArray = dataArray.map(() => "0");
-  const targetArray = dataArray.map(() => voting.address);
+  const valuesArray = retrievableRewards.map(() => "0");
+  const targetArray = retrievableRewards.map(({ voting }) => voting.options.address);
 
   const transactionBatcher = await TransactionBatcher.at(argv.batcherAddress);
-  const txn = transactionBatcher.contract.methods.batchSend(targetArray, valuesArray, dataArray);
-  const account = (await web3.eth.getAccounts())[0];
-  const gasEstimate = await txn.estimateGas({ from: account });
 
-  if (gasEstimate > 9000000) {
-    throw "The transaction requires too much gas. Will need to be split up.";
-  }
+  const gasEstimator = new GasEstimator(
+    winston.createLogger({
+      silent: true
+    }),
+    60, // Time between updates.
+    100 // Default gas price.
+  );
+
+  // These nested calls just chunk up the above arrays into 30 transaction chunks, then organize them in groups for
+  // sending on-chain.
+  await Promise.all(
+    lodash
+      .zip(...[targetArray, valuesArray, dataArray].map(arr => lodash.chunk(arr, 30)))
+      .map(async ([chunkedTargetArray, chunkedValueArray, chunkedDataArray]) => {
+        const txn = transactionBatcher.contract.methods.batchSend(
+          chunkedTargetArray,
+          chunkedValueArray,
+          chunkedDataArray
+        );
+        const gasEstimate = await txn.estimateGas({ from: account });
 
-  await transactionBatcher.batchSend(targetArray, valuesArray, dataArray);
+        if (gasEstimate > 6000000) {
+          throw "The transaction requires too much gas. Will need to be split up.";
+        }
+        await gasEstimator.update();
+
+        const receipt = await txn.send({
+          gasPrice: gasEstimator.getCurrentFastPrice(),
+          gas: gasEstimate * 2,
```

Comment:
*2 seems potentially dangerous if this script is run during high gas times, but at least it will execute! Would be cool to add a quick metamask style prompt here for user to select between standard, fast, and instant gas: https://www.digitalocean.com/community/tutorials/nodejs-interactive-command-line-prompts

Perhaps something we should add as a common lib and import into all mainnet scripts

Reply:
yea i agree this could get really crazy. would be nice maybe specify the gas as instant, fast, normal as a cmd line arg when running script.

Reply:
Note: this is the _gas_ to send, not the price. Not sure if we should multiply it by this much. I mainly did this because I'm not sure about how refunds are handled and these rewards transactions get a lot of refunds. Probably deserves a little more research.